### ISSUE TITLE
Conv1d Version?

### ISSUE THREAD

**Issue by kayuksel** [joe-siyuan-qiao/WeightStandardization#2]
Hello, is the following Conv1d version correct? Because I have tried it with a 1d convolutional layer that is followed by a batch-norm. It unfortunately did not improve the existing results but made them worse.
```
class Conv1d(nn.Conv1d):
    def __init__(self, in_channels, out_channels, kernel_size, stride=1,
                 padding=0, dilation=1, groups=1, bias=True):
        super(Conv2d, self).__init__(in_channels, out_channels, kernel_size, stride,
                 padding, dilation, groups, bias)
    def forward(self, x):
        weight = self.weight
        weight_mean = weight.mean(dim=1, keepdim=True).mean(dim=2, keepdim=True)
        weight = weight - weight_mean
        std = weight.view(weight.size(0), -1).std(dim=1).view(-1, 1, 1) + 1e-5
        weight = weight / std.expand_as(weight)
        return F.conv1d(x, weight, self.bias, self.stride,
                        self.padding, self.dilation, self.groups)
```

**Comment by joe-siyuan-qiao** [comment]
Hi kayuksel,

Thanks for the question.

We do not have the implementation for Conv1d as it is not used in our experiments, but I guess it would look similar to yours. In your case, super(Conv2d, self) in the __init__ should be super(Conv1d, self). But we are not sure if this will solve all the problem. Please update after this is fixed. Thanks!

Siyuan

**Comment by kayuksel** [comment]
Sorry actually I have fixed that while running it. However, I couldnt
observe any benefit at least in the problem that I have experimented with.

**Comment by joe-siyuan-qiao** [comment]
Ok, I see. Would you mind sharing the batch size per GPU, and the performances of using GN, GN+WS, BN, and BN+WS? Because if the batch size is large enough, WS may not be able to help BN -- it will help GN, though. Another reason might be that replacing layers that do not have BN/GN directly following them may harm the training, but this may not apply to your case.

**Comment by kayuksel** [comment]
I am using a single gpu with batch-size 512. All convolutional layers are
directly followed with BN, Dropout and RReLU.

**Comment by joe-siyuan-qiao** [comment]
Got it. Then, I think maybe WS is not a good choice for your case because your batch size is 512, which is large enough to train a good BN. WS is majorly targeted at accelerating micro-batch training where batch sizes are typically 1-2. WS may not be able to help large-batch training for networks with BN.
